apps/nova/src/pages/UpcomingLessons.tsx

```diff
@@ -21,10 +55,35 @@ const UpcomingLessons: React.FC = () => {
     canceled: true,
   });
 
+  // useMutation hooks
+  const { mutate: createRating } = useCreateRatingTutor({
+    onSuccess: () => setRateDialogInfo((prev) => ({ ...prev, enabled: false })),
+    onError: () => console.log("createRating failed!"),
+  });
+
+  const { mutate: editRating } = useEditRatingTutor({
+    onSuccess: () => setRateDialogInfo((prev) => ({ ...prev, enabled: false })),
+    onError: () => console.log("editRating failed!"),
+  });
+
+  // useEffect hooks
   useEffect(() => {
     if (!user) return navigate(Route.Root);
   }, [navigate, user]);
 
+  useEffect(() => {
+    if (user?.role !== "student") return;
+    const { tutorId, lessonId } = getLessonQuery(params);
+    setRateDialogInfo((prev) => ({ ...prev, tutorId, lessonId }));
+  }, [params, user?.role]);
+
+  useEffect(() => {
+    if (!ratingQuery.data || !user) return;
+    if (ratingQuery.data.list[0].userId === user.id) return;
+    const enabled = !!(rateDialogInfo.tutorId && rateDialogInfo.lessonId);
+    setRateDialogInfo((prev) => ({ ...prev, enabled }));
+  }, [ratingQuery.data, rateDialogInfo.lessonId, rateDialogInfo.tutorId, user]);
```

Comment:
I believe we can improve the code to make it error-proof and more clean. 

```suggestion
  useEffect(() => {
    if (!ratingQuery.data || ratingQuery.isLoading || ratingQuery.isPending || !user) return;
    const rating = first(ratingQuery.data)
    const enabled = rating && rating.userId === user.id; 
    setRateDialogInfo((prev) => ({ ...prev, enabled }));
  }, [ratingQuery.data, rateDialogInfo.lessonId, rateDialogInfo.tutorId, user]);
```